### PR TITLE
adding option to just export ipa from archive_path

### PR DESCRIPTION
**Pull request by dreyhomedev**
Added an option to only export an ipa file from a xarchive.

For me it is required to export development, adhoc and appstore version of an build. This option enables me to skip two build processes and only export ipas from the same build archive.

### CODE REVIEW

**Comment by dreyhomedev** [comment]
This addresses #12

**Comment by dreyhomedev** [comment]
Don't know why the build failed. No concrete error was reported from travis. What can I do to rerun the build?

**Comment by lacostej** [comment]
Try pushing again :) Sometimes it fails like that.

**Comment by dreyhomedev** [comment]
Already tried a second time but same result?
I looked into the travis output and it hangs just after displaying the help of gym.
I'l try a third time... just to be sure ;-)

File: lib/gym/runner.rb [hunk 1]
```diff
@@ -9,7 +9,9 @@ class Runner
     def run
       clear_old_files
```
Comment (lacostej) [inline lib/gym/runner.rb]: in the case of use of `:use_legacy_build_api` option, the `clear_old_files` will delete the previously built archive. So the export only will then fail. I suspect the `clear_old_files` should move into the `unless` block as well

**Comment by dreyhomedev** [comment]
Thanks @lacostej . I missed that.
But still no success. :sob:

**Comment by dreyhomedev** [comment]
I think this may be related to #119

**Comment by dreyhomedev** [comment]
@lacostej Is there a way to run the tests of travis on my own machine? Maybe I can get a hold on the problem.

**Comment by lacostej** [comment]
@dreyhomedev you could try `rake test` but it doesn't fail on my machine as the problem is CI related.

It is most probably related to `enable_paging` in commander 

https://github.com/tj/commander/blob/d10fc03ce9d44a0a34bb913df99dded4a0bd5287/lib/commander/user_interaction.rb#L274

The `| cat` [was added just a few days](https://github.com/fastlane/fastlane/pull/1349) ago by @mfurtak, so maybe he or @KrauseFx  knows.

**Comment by lacostej** [comment]
I sent fastlane/fastlane#1378, maybe that helps

**Comment by dreyhomedev** [comment]
Thanks a lot @lacostej. I tried to run `rake test` but since i'm not a ruby dev I'm lacking the ruby environment to run rake. ^^
Is there a complete environment I can install wich contains all needed resources?

I really appreciate your help!

**Comment by dreyhomedev** [comment]
Hey @lacostej I saw your PR was merged. Do you think i should try again with a new pull? ;-)

**Comment by lacostej** [comment]
@dreyhomedev a mine didn't work fully so a few more things were tried (e.g. https://github.com/fastlane/fastlane/commit/4226f896fa59e0d5507a5c2dfed2812f24b8d741), so yes I think you can try with a new push.

**Comment by dreyhomedev** [comment]
YAY!!

File: lib/gym/options.rb [hunk 2]
```diff
@@ -122,7 +122,11 @@ def self.plain_options
                                      conflict_block: proc do |value|
                                        UI.user_error!("'#{value.key}' must be false to use 'export_options'")
                                      end),
-
+        FastlaneCore::ConfigItem.new(key: :just_export,
+                                     env_name: "GYM_JUST_EXPORT",
+                                     description: "Export ipa from previously build xarchive. Uses archive_path as source",
+                                     is_string: false,
```
Comment (lacostej) [inline lib/gym/options.rb]: are there options this new option should conflict with ?

File: lib/gym/options.rb [hunk 3]
```diff
@@ -122,7 +122,11 @@ def self.plain_options
                                      conflict_block: proc do |value|
                                        UI.user_error!("'#{value.key}' must be false to use 'export_options'")
                                      end),
-
+        FastlaneCore::ConfigItem.new(key: :just_export,
```
Comment (lacostej) [inline lib/gym/options.rb]: I wonder if it would be more natural to have a new command. like `gym export`

File: lib/gym/error_handler.rb [hunk 4]
```diff
@@ -99,6 +99,7 @@ def handle_empty_archive
         print "For more information visit https://developer.apple.com/library/ios/technotes/tn2215/_index.html"
         print "Also, make sure to have a valid code signing identity and provisioning profile installed"
         print "Follow this guide to setup code signing https://github.com/fastlane/fastlane/blob/master/docs/CodeSigning.md"
+        print "If your intension was only to export an ipa be sure to provide a valid archive at the archive path."
```
Comment (lacostej) [inline lib/gym/error_handler.rb]: typo. intention ? goal ?

**Comment by dreyhomedev** [comment]
I do not see any conflicts with the current options. It will fail if you do not specify a build archive or the archive is not at the specified path.

To have a new command would also be nice, but i'm not ruby pro enough to implement that  :wink: 
How would one specify it in the Fastfile? Just `export`?

File: lib/gym/error_handler.rb (hunk 4 above)
Comment (dreyhomedev) [inline lib/gym/error_handler.rb]: Damn it... I'll fix that. Must be intention.

**Comment by lacostej** [comment]
For the new command, you need to look at

https://github.com/dreyhomedev/gym/blob/master/lib/gym/commands_generator.rb#L36-L44

But let's see what other think before you do it. Maybe the export parameter is good enough ?

**Comment by dreyhomedev** [comment]
OK, I'll ask the guys in the issue #12

File: lib/gym/options.rb (hunk 3 above)
Comment (czechboy0) [inline lib/gym/options.rb]: :+1:

**Comment by mokagio** [comment]
Hey @dreyhomedev, great work.

I usually see command line options like `--no-xxx` in CLIs that allow to modify the standard behaviour. For example CocoaPods `pod install` provides `--no-clean`, `--no-integrate`, and `--no-repo-update`. Carthage `carthage update` provides `--no-build`, `--no-checkout`, `--no-use-binaries`. Git `git commit` provides `--no-pager`, `--no-replace-objects`. And so on.

What do you think of the idea of making your option a `--no-sign` instead?

**Comment by czechboy0** [comment]
Also `--no-build` I think

**Comment by dreyhomedev** [comment]
@mokagio Thanks for the input.
I admit that `just_export` doesn't sound very professional but I've not seen any option in the fastlane tools that is names like your suggestions. Instead I saw `skip_build` in scan. I would prefer to name it `skip_build_archive` to be consistent among the fastlane tools.

How about that?

**Comment by mpirri** [comment]
@dreyhomedev Looks awesome!  Merging.  :+1:

**Comment by czechboy0** [comment]
:+1:

**Pull request merged**